### DISCUSSION TITLE
hello

### DISCUSSION THREAD

**Discussion by Patrick197030100** [Westcott1/Waveshare-ESP32-S3-Touch-LCD-4.3-and-Arduino#9] [General]
INFO ESPHome 2024.7.3
INFO Reading configuration /config/esphome/yambms-lcd.yaml...
INFO Updating https://github.com/txubelaxu/esphome-jk-bms.git@main
INFO Detected timezone 'Europe/Paris'
Failed config

display.rpi_dpi_rgb: [source /config/esphome/packages/board/board_teste-lcd.yaml:20]
  
  Pin 40 is used in multiple places.
  platform: rpi_dpi_rgb
  id: rpi_disp
  dimensions: 
    width: 800
    height: 480
    offset_height: 0
    offset_width: 0
  de_pin: 
    number: 40
    mode: 
      output: True
      input: False
output.gpio: [source /config/esphome/packages/board/board_teste-lcd.yaml:61]
  
  Pin 17 is used in multiple places.
  platform: gpio
  id: esp_led
  pin: 
    number: 17
    mode: 
      output: True
      input: False
      open_drain: False
      pullup: False
      pulldown: False
    inverted: False
    ignore_pin_validation_error: False
spi: [source /config/esphome/packages/board/board_teste-lcd.yaml:95]
  
  Pin 7 is used in multiple places.
  - clk_pin: 
      number: 7
      mode: 
        output: True
        input: False
        open_drain: False
        pullup: False
        pulldown: False
      inverted: False
      ignore_pin_validation_error: False
      ignore_strapping_warning: False
      drive_strength: 20.0
Pin 40 is used in multiple places

Pin 39 is used in multiple places

Pin 39 is used in multiple places

Pin 41 is used in multiple places

Pin 41 is used in multiple places

Pin 7 is used in multiple places

Pin 17 is used in multiple places
hello I am looking for information on implementing the lvgl esphome because it is impossible to display the screen with my yaml!!! thank you

Icône de validation par la communauté
![IMG_5178](https://github.com/user-attachments/assets/3bf73ba4-adcc-4328-b8fb-916fbceb5ed6)
![IMG_5066](https://github.com/user-attachments/assets/17f7bff2-73dd-49ac-9f0b-48376d9d78b0)
because I want to put the atom file on my waveshare lcd touch 7

**Comment by Patrick197030100** [comment]
substitutions:
  board_chip: "ESP32-S3"
  board_name: "lcd-touch"

esp32:
  board: esp32-s3-devkitc-1
  framework:
    type: esp-idf
    sdkconfig_options:
      CONFIG_ESP32S3_DEFAULT_CPU_FREQ_240: y
      CONFIG_ESP32S3_DATA_CACHE_64KB: y
      CONFIG_SPIRAM_FETCH_INSTRUCTIONS: y
      CONFIG_SPIRAM_RODATA: y

esphome:
  name: my_device
  platformio_options:
    board_build.flash_mode: dio
display:
  - platform: rpi_dpi_rgb
    id: rpi_disp
    dimensions:
      width: 800
      height: 480
    de_pin: 40
    hsync_pin: 39
    vsync_pin: 41
    pclk_pin: 07
    
    data_pins:
      red:
        - GPIO1
        - GPIO2
        - GPIO42
        - GPIO41
        - GPIO40
      green:
        - GPIO39
        - GPIO0
        - GPIO45
        - GPIO48
        - GPIO47
        - GPIO21
      blue:
        - GPIO14
        - GPIO38
        - GPIO18
        - GPIO17
        - GPIO10
external_components:
  - source: github://esphome/esphome#dev
    components:
      - lvgl

psram:
  mode: octal
  speed: 80MHz

output:
  - id: esp_led
    platform: gpio
    pin: 17 

logger:
  hardware_uart: USB_SERIAL_JTAG

uart:
  - id: uart_esp_1
    tx_pin: GPIO43
    rx_pin: GPIO44
    baud_rate: 115200

canbus:
  - platform: esp32_can
    id: canbus_node1
    tx_pin: GPIO20
    rx_pin: GPIO19
    can_id: 1
    bit_rate: 500kbps

i2c:
  sda: GPIO8
  scl: GPIO9
  scan: True
  id: bus_a

light:
  - platform: binary
    id: esp_light
    output: esp_led
    name: "yambms On board light"
    internal: True
    disabled_by_default: False
    restore_mode: ALWAYS_OFF
spi:
  clk_pin: GPIO07
  mosi_pin: GPIO11
  miso_pin: GPIO12  # Include this pin only if needed by your display
  id: spi_bus


is no good help my than you !

**Comment by Westcott1** [comment]
Hi Patrick197030100,
Thanks for posting.
I'm not sure what it is that you are asking - I have no knowledge of ESPhome at all.
However I am VERY interested in how the CAN/RS485 works, and shows 98% SOC
